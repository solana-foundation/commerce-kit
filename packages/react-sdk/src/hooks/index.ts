/**
 * Checkout Hooks
 * Export all checkout-related hooks for state management
 */

// Core business logic hooks
export { useSolanaPay } from './use-solana-pay';
export { useTipForm } from './use-tip-form';

export { useAnimationStyles } from './use-animation-styles';
export { useCopyToClipboard } from './use-copy-to-clipboard';

// UI utility hooks
export { useHover } from './use-hover';
export { useThemeStyles } from './use-theme-styles';
export { usePaymentStatus } from './use-payment-status';
export { useTimer } from './use-timer';
export { useDropdown } from './use-dropdown';
export { useAsync } from './use-async';
export { useFormField } from './use-form-field';