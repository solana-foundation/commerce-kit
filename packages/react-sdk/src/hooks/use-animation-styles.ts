/**
 * Animation Styles Hook
 * Manages injection of animation styles
 */

import { useEffect } from 'react';
import { ANIMATION_STYLES } from '../constants/tip-modal';

const STYLE_ID = 'sc-tip-modal-anim';

export function useAnimationStyles() {
  useEffect(() => {
    if (typeof document !== 'undefined' && !document.getElementById(STYLE_ID)) {
      const styleEl = document.createElement('style');
      styleEl.id = STYLE_ID;
      styleEl.textContent = ANIMATION_STYLES;
      document.head.appendChild(styleEl);
    }
  }, []);
}
