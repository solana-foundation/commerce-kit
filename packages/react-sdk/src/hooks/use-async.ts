/**
 * useAsync Hook
 * Handle async operations with loading, error, and data states
 */

import { useState, useCallback, useEffect, useRef } from 'react';

interface UseAsyncState<T> {
  data: T | null;
  error: Error | null;
  loading: boolean;
}

interface UseAsyncReturn<T> extends UseAsyncState<T> {
  execute: (...args: any[]) => Promise<T | null>;
  reset: () => void;
}

export function useAsync<T = any>(
  asyncFunction?: (...args: any[]) => Promise<T>,
  immediate: boolean = false
): UseAsyncReturn<T> {
  const [state, setState] = useState<UseAsyncState<T>>({
    data: null,
    error: null,
    loading: false,
  });

  // Keep track of the current async operation to prevent race conditions
  const asyncRef = useRef<number>(0);

  const execute = useCallback(async (...args: any[]): Promise<T | null> => {
    if (!asyncFunction) return null;

    // Increment counter for this execution
    const currentExecution = ++asyncRef.current;

    setState(prev => ({ ...prev, loading: true, error: null }));

    try {
      const result = await asyncFunction(...args);
      
      // Only update state if this is still the current execution
      if (currentExecution === asyncRef.current) {
        setState({
          data: result,
          error: null,
          loading: false,
        });
      }
      
      return result;
    } catch (error) {
      // Only update state if this is still the current execution
      if (currentExecution === asyncRef.current) {
        setState({
          data: null,
          error: error instanceof Error ? error : new Error(String(error)),
          loading: false,
        });
      }
      
      return null;
    }
  }, [asyncFunction]);

  const reset = useCallback(() => {
    // Increment counter to cancel any pending operations
    asyncRef.current++;
    setState({
      data: null,
      error: null,
      loading: false,
    });
  }, []);

  // Execute immediately if requested
  useEffect(() => {
    if (immediate && asyncFunction) {
      execute();
    }
  }, [immediate, asyncFunction, execute]);

  return {
    ...state,
    execute,
    reset,
  };
}
