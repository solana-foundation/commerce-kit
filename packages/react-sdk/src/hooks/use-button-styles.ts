/**
 * Button Styles Hook
 * Provides optimized styling for interactive buttons
 */

import { useMemo, useState, useCallback } from 'react';
import { getButtonShadow, getButtonBorder, getAccessibleTextColor, getRadius } from '../utils';
import type { ThemeConfig } from '../types';

interface ButtonStyleHookOptions {
  theme: Required<ThemeConfig>;
  isDisabled?: boolean;
  variant?: 'action' | 'selection';
  isSelected?: boolean;
}

export function useButtonStyles({ 
  theme, 
  isDisabled = false, 
  variant = 'action',
  isSelected = false 
}: ButtonStyleHookOptions) {
  const [isHovered, setIsHovered] = useState(false);
  const [isPressed, setIsPressed] = useState(false);
  const [isFocused, setIsFocused] = useState(false);

  // Base styles computation
  const styles = useMemo(() => {
    const borderStyle = (() => {
      const b = getButtonBorder(theme);
      return b === 'none' ? '1.5px solid transparent' : b;
    })();

    if (variant === 'action') {
      return {
        width: '100%',
        padding: '1rem',
        backgroundColor: isDisabled 
          ? '#9ca3af' 
          : isHovered 
            ? theme.secondaryColor 
            : theme.primaryColor,
        color: isDisabled 
          ? 'white' 
          : getAccessibleTextColor(isHovered ? theme.secondaryColor : theme.primaryColor),
        border: isDisabled ? '1.5px solid transparent' : borderStyle,
        borderRadius: getRadius('button', theme.borderRadius),
        fontSize: '1rem',
        fontWeight: '600',
        cursor: isDisabled ? 'not-allowed' : 'pointer',
        transition: 'background-color 0.2s ease, box-shadow 0.2s ease, transform 0.05s ease',
        fontFamily: theme.fontFamily,
        boxShadow: isDisabled 
          ? 'none'
          : (isHovered || isFocused)
            ? `${getButtonShadow(theme.buttonShadow)}, 0 0 0 4px rgba(202, 202, 202, 0.45)`
            : getButtonShadow(theme.buttonShadow),
        transform: isPressed ? 'scale(0.97)' : 'scale(1)',
        outlineOffset: 2,
      } as React.CSSProperties;
    }

    // Selection variant (for preset amounts, payment methods, etc.)
    return {
      border: isSelected ? `3px solid #ffffff` : '1px solid #e5e7eb',
      borderRadius: getRadius('preset', theme.borderRadius),
      backgroundColor: isSelected ? `${theme.primaryColor}10` : '#ffffff',
      color: isSelected ? 'rgba(0, 0, 0, 0.8)' : theme.textColor,
      fontSize: '19px',
      fontWeight: '400',
      cursor: 'pointer',
      transition: 'all 0.2s, transform 0.1s ease',
      transform: isPressed ? 'scale(0.98)' : 'scale(1)',
      boxShadow: isSelected 
        ? `0 0 0 2px ${theme.primaryColor}60` 
        : '0 1px 2px 0 rgba(0, 0, 0, 0.05)',
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center'
    } as React.CSSProperties;
  }, [theme, isDisabled, isHovered, isFocused, variant, isSelected, isPressed]);

  // Event handlers
  const handlers = {
    onMouseEnter: useCallback(() => {
      if (!isDisabled) setIsHovered(true);
    }, [isDisabled]),

    onMouseLeave: useCallback(() => {
      setIsHovered(false);
      setIsPressed(false);
    }, [setIsPressed]),

    onMouseDown: useCallback((e: React.MouseEvent<HTMLButtonElement>) => {
      if (!isDisabled) {
        setIsPressed(true);
      }
    }, [isDisabled, setIsPressed, variant]),

    onMouseUp: useCallback(() => {
      setIsPressed(false);
    }, [setIsPressed]),

    onTouchEnd: useCallback(() => {
      setIsPressed(false);
    }, [setIsPressed]),

    onFocus: useCallback(() => {
      if (!isDisabled && variant === 'action') {
        setIsFocused(true);
      }
    }, [isDisabled, variant]),

    onBlur: useCallback(() => {
      if (variant === 'action') {
        setIsFocused(false);
      }
    }, [variant]),
  };

  return {
    styles,
    handlers,
    isHovered,
    isFocused,
  };
}
