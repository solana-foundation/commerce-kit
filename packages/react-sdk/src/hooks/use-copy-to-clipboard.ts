import { useState } from 'react';

interface UseCopyToClipboardReturn {
  copied: boolean;
  isHovered: boolean;
  setIsHovered: (hovered: boolean) => void;
  copyToClipboard: (text: string) => Promise<void>;
}

export function useCopyToClipboard(): UseCopyToClipboardReturn {
  const [copied, setCopied] = useState(false);
  const [isHovered, setIsHovered] = useState(false);

  const copyToClipboard = async (text: string) => {
    try {
      await navigator.clipboard.writeText(text);
      setCopied(true);
      setIsHovered(false);
      setTimeout(() => {
        setCopied(false);
        setIsHovered(false);
      }, 2000);
    } catch (err) {
      // Fallback for older browsers
      const textArea = document.createElement('textarea');
      textArea.value = text;
      document.body.appendChild(textArea);
      textArea.select();
      document.execCommand('copy');
      document.body.removeChild(textArea);
      setCopied(true);
      setIsHovered(false);
      setTimeout(() => {
        setCopied(false);
        setIsHovered(false);
      }, 2000);
    }
  };

  return {
    copied,
    isHovered,
    setIsHovered,
    copyToClipboard
  };
}
