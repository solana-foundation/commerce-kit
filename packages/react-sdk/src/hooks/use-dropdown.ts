/**
 * useDropdown Hook
 * Reusable dropdown/select state management
 */

import { useState, useCallback, useRef, useEffect } from 'react';
import type { RefObject } from 'react';

interface UseDropdownOptions {
  initialOpen?: boolean;
  closeOnSelect?: boolean;
  closeOnClickOutside?: boolean;
}

interface UseDropdownReturn<T = any, E extends HTMLElement = HTMLElement> {
  isOpen: boolean;
  selectedValue: T | null;
  open: () => void;
  close: () => void;
  toggle: () => void;
  select: (value: T) => void;
  clear: () => void;
  ref: RefObject<E | null>;
}

export function useDropdown<T = any, E extends HTMLElement = HTMLElement>({
  initialOpen = false,
  closeOnSelect = true,
  closeOnClickOutside = true
}: UseDropdownOptions = {}): UseDropdownReturn<T, E> {
  const [isOpen, setIsOpen] = useState(initialOpen);
  const [selectedValue, setSelectedValue] = useState<T | null>(null);
  const ref = useRef<E | null>(null);

  const open = useCallback(() => {
    setIsOpen(true);
  }, []);

  const close = useCallback(() => {
    setIsOpen(false);
  }, []);

  const toggle = useCallback(() => {
    setIsOpen(prev => !prev);
  }, []);

  const select = useCallback((value: T) => {
    setSelectedValue(value);
    if (closeOnSelect) {
      setIsOpen(false);
    }
  }, [closeOnSelect]);

  const clear = useCallback(() => {
    setSelectedValue(null);
  }, []);

  // Handle click outside
  useEffect(() => {
    if (!closeOnClickOutside || !isOpen) return;

    const handleClickOutside = (event: MouseEvent) => {
      if (ref.current && !ref.current.contains(event.target as Node)) {
        setIsOpen(false);
      }
    };

    document.addEventListener('mousedown', handleClickOutside);
    return () => document.removeEventListener('mousedown', handleClickOutside);
  }, [isOpen, closeOnClickOutside]);

  // Handle escape key
  useEffect(() => {
    if (!isOpen) return;

    const handleKeyDown = (event: KeyboardEvent) => {
      if (event.key === 'Escape') {
        setIsOpen(false);
      }
    };

    document.addEventListener('keydown', handleKeyDown);
    return () => document.removeEventListener('keydown', handleKeyDown);
  }, [isOpen]);

  return {
    isOpen,
    selectedValue,
    open,
    close,
    toggle,
    select,
    clear,
    ref,
  };
}
