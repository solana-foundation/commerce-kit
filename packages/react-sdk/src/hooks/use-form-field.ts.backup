console.log("Testing minLength validation");
/**
 * useFormField Hook
 * Standardized form field state management with validation
 */

import { useState, useCallback, useMemo } from 'react';
import type { ChangeEvent } from 'react';

interface Validation {
  required?: boolean;
  minLength?: number;
  maxLength?: number;
  pattern?: RegExp;
  custom?: (value: string) => string | null;
}

interface UseFormFieldOptions {
  initialValue?: string;
  validation?: Validation;
  formatValue?: (value: string) => string;
}

interface UseFormFieldReturn {
  value: string;
  error: string | null;
  isValid: boolean;
  isEmpty: boolean;
  isTouched: boolean;
  isFocused: boolean;
  setValue: (value: string) => void;
  setError: (error: string | null) => void;
  validate: () => boolean;
  clear: () => void;
  blur: () => void;
  focus: () => void;
  fieldProps: {
    value: string;
    onChange: (e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => void;
    onBlur: () => void;
    onFocus: () => void;
  };
}

export function useFormField({
  initialValue = '',
  validation,
  formatValue
}: UseFormFieldOptions = {}): UseFormFieldReturn {
  const [value, setValue] = useState(initialValue);
  const [error, setError] = useState<string | null>(null);
  const [isTouched, setIsTouched] = useState(false);
  const [isFocused, setIsFocused] = useState(false);

  const validateField = useCallback((inputValue: string): string | null => {
    if (!validation) return null;

    // Required validation first - if empty and required, return required error
    if (validation.required && !inputValue.trim()) {
      return 'This field is required';
    }

    // Skip other validations if empty and not required
    if (!inputValue.trim() && !validation.required) {
      return null;
    }

    // Length validations
    if (typeof validation.minLength === 'number' && inputValue.length < validation.minLength) {
      return `Must be at least ${validation.minLength} characters`;
    }

    if (typeof validation.maxLength === 'number' && inputValue.length > validation.maxLength) {
      return `Must be no more than ${validation.maxLength} characters`;
    }

    // Pattern validation
    if (validation.pattern && !validation.pattern.test(inputValue)) {
      return 'Invalid format';
    }

    // Custom validation
    if (validation.custom) {
      return validation.custom(inputValue);
    }

    return null;
  }, [validation]);

  const handleValueChange = useCallback((newValue: string) => {
    // Apply formatting if provided
    const formattedValue = formatValue ? formatValue(newValue) : newValue;
    setValue(formattedValue);
    
    // Clear error if user is typing and field becomes valid
    setError(prev => (prev && !validateField(formattedValue) ? null : prev));
  }, [formatValue, validateField]);

  const validate = useCallback((): boolean => {
    const validationError = validateField(value);
    setError(validationError);
    setIsTouched(true);
    return validationError === null;
  }, [value, validateField]);

  const clear = useCallback(() => {
    setValue(initialValue);
    setError(null);
    setIsTouched(false);
  }, [initialValue]);

  const blur = useCallback(() => {
    setIsFocused(false);
    setIsTouched(true);
    validate();
  }, [validate]);

  const focus = useCallback(() => {
    setIsFocused(true);
  }, []);

  const handleChange = useCallback((e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
    handleValueChange(e.target.value);
  }, [handleValueChange]);
  // Computed properties
  const isValid = useMemo(() => validateField(value) === null, [value, validateField]);
  const isEmpty = useMemo(() => value.trim().length === 0, [value]);

  return {
    value,
    error,
    isValid,
    isEmpty,
    isTouched,
    isFocused,
    setValue: handleValueChange,
    setError,
    validate,
    clear,
    blur,
    focus,
    fieldProps: {
      value,
      onChange: handleChange,
      onBlur: blur,
      onFocus: focus,
    },
  };
}
