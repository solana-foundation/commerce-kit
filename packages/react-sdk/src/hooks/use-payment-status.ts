/**
 * usePaymentStatus Hook
 * Centralized payment status management
 */

import { useState, useCallback } from 'react';

export type PaymentStatus =
    | 'idle'
    | 'connecting'
    | 'scanning'
    | 'processing'
    | 'confirming'
    | 'success'
    | 'error'
    | 'cancelled'
    | 'timeout';

interface UsePaymentStatusReturn {
    status: PaymentStatus;
    error: string | null;
    isLoading: boolean;
    isComplete: boolean;
    canRetry: boolean;
    setStatus: (status: PaymentStatus) => void;
    setError: (error: string | null) => void;
    handleSuccess: () => void;
    handleError: (error: string | Error) => void;
    handleCancel: () => void;
    handleTimeout: () => void;
    retry: () => void;
    reset: () => void;
}

export function usePaymentStatus(): UsePaymentStatusReturn {
    const [status, setStatus] = useState<PaymentStatus>('idle');
    const [error, setError] = useState<string | null>(null);

    const handleSuccess = useCallback(() => {
        setStatus('success');
        setError(null);
    }, []);

    const handleError = useCallback((errorInput: string | Error) => {
        setStatus('error');
        const errorMessage = errorInput instanceof Error ? errorInput.message : errorInput;
        setError(errorMessage);
    }, []);

    const handleCancel = useCallback(() => {
        setStatus('cancelled');
        setError(null);
    }, []);

    const handleTimeout = useCallback(() => {
        setStatus('timeout');
        setError('Payment request timed out. Please try again.');
    }, []);

    const retry = useCallback(() => {
        setStatus('idle');
        setError(null);
    }, []);

    const reset = useCallback(() => {
        setStatus('idle');
        setError(null);
    }, []);

    // Computed properties
    const isLoading = ['connecting', 'scanning', 'processing', 'confirming'].includes(status);
    const isComplete = ['success', 'error', 'cancelled', 'timeout'].includes(status);
    const canRetry = ['error', 'timeout'].includes(status);

    return {
        status,
        error,
        isLoading,
        isComplete,
        canRetry,
        setStatus,
        setError,
        handleSuccess,
        handleError,
        handleCancel,
        handleTimeout,
        retry,
        reset,
    };
}
