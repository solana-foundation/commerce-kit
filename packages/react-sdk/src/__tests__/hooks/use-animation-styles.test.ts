import { describe, it, expect, vi, beforeEach, afterEach } from 'vitest';
import { renderHook } from '@testing-library/react';
import { useAnimationStyles } from '../../hooks/use-animation-styles';

// Mock the constants
vi.mock('../../constants/tip-modal', () => ({
    ANIMATION_STYLES: `
@keyframes sc-tip-modal-slide-up {
  0% { transform: translateY(16px); opacity: 0; }
  100% { transform: translateY(0); opacity: 1; }
}
.sc-tip-modal-anim {
  transition: height 500ms cubic-bezier(0.4, 0, 0.2, 1) !important;
}
@media (prefers-reduced-motion: reduce) {
  .sc-tip-modal-anim { 
    animation: none !important;
    transition: none !important;
  }
}`.trim(),
}));

describe('useAnimationStyles', () => {
    const STYLE_ID = 'sc-tip-modal-anim';

    beforeEach(() => {
        // Clean up any existing style elements
        const existingStyle = document.getElementById(STYLE_ID);
        if (existingStyle) {
            existingStyle.remove();
        }
    });

    afterEach(() => {
        // Clean up after each test
        const styleEl = document.getElementById(STYLE_ID);
        if (styleEl) {
            styleEl.remove();
        }
    });

    it('should inject animation styles into document head', () => {
        // Initially no style element should exist
        expect(document.getElementById(STYLE_ID)).toBe(null);

        // Render the hook
        renderHook(() => useAnimationStyles());

        // Style element should now be present
        const styleEl = document.getElementById(STYLE_ID);
        expect(styleEl).toBeTruthy();
        expect(styleEl?.tagName).toBe('STYLE');
        expect(styleEl?.textContent).toContain('@keyframes sc-tip-modal-slide-up');
        expect(styleEl?.textContent).toContain('sc-tip-modal-anim');
        expect(styleEl?.textContent).toContain('prefers-reduced-motion');
    });

    it('should not duplicate style element if already exists', () => {
        // Pre-create a style element
        const existingStyle = document.createElement('style');
        existingStyle.id = STYLE_ID;
        existingStyle.textContent = 'existing styles';
        document.head.appendChild(existingStyle);

        // Render the hook
        renderHook(() => useAnimationStyles());

        // Should still only be one style element
        const styleElements = document.querySelectorAll(`#${STYLE_ID}`);
        expect(styleElements.length).toBe(1);

        // Should preserve the existing content (not overwrite)
        expect(styleElements[0]?.textContent).toBe('existing styles');
    });

    it('should handle multiple hook instances gracefully', () => {
        // Render the hook multiple times
        const { unmount: unmount1 } = renderHook(() => useAnimationStyles());
        const { unmount: unmount2 } = renderHook(() => useAnimationStyles());
        const { unmount: unmount3 } = renderHook(() => useAnimationStyles());

        // Should still only be one style element
        const styleElements = document.querySelectorAll(`#${STYLE_ID}`);
        expect(styleElements.length).toBe(1);
        expect(styleElements[0]?.textContent).toContain('@keyframes');

        // Clean up
        unmount1();
        unmount2();
        unmount3();
    });

    it('should work in browser environment with document', () => {
        // Verify document is available (should be with happy-dom)
        expect(typeof document).toBe('object');
        expect(document.getElementById).toBeDefined();
        expect(document.createElement).toBeDefined();
        expect(document.head).toBeDefined();

        renderHook(() => useAnimationStyles());

        const styleEl = document.getElementById(STYLE_ID);
        expect(styleEl).toBeTruthy();
        expect(document.head.contains(styleEl)).toBe(true);
    });

    it('should inject complete animation styles with proper formatting', () => {
        renderHook(() => useAnimationStyles());

        const styleEl = document.getElementById(STYLE_ID);
        expect(styleEl?.textContent).toMatchSnapshot();

        // Verify specific animation properties
        expect(styleEl?.textContent).toContain('transform: translateY(16px)');
        expect(styleEl?.textContent).toContain('opacity: 0');
        expect(styleEl?.textContent).toContain('transform: translateY(0)');
        expect(styleEl?.textContent).toContain('opacity: 1');
        expect(styleEl?.textContent).toContain('transition: height 500ms cubic-bezier(0.4, 0, 0.2, 1)');
        expect(styleEl?.textContent).toContain('animation: none !important');
        expect(styleEl?.textContent).toContain('transition: none !important');
    });

    it('should not interfere with existing head content', () => {
        // Add some existing content to head
        const existingMeta = document.createElement('meta');
        existingMeta.name = 'test';
        existingMeta.content = 'existing';
        document.head.appendChild(existingMeta);

        const initialHeadChildCount = document.head.children.length;

        renderHook(() => useAnimationStyles());

        // Should have added exactly one more element
        expect(document.head.children.length).toBe(initialHeadChildCount + 1);

        // Existing content should still be there
        expect(document.head.querySelector('meta[name="test"]')).toBeTruthy();

        // New style should be there too
        expect(document.getElementById(STYLE_ID)).toBeTruthy();
    });

    it('should handle hook unmount without removing styles', () => {
        const { unmount } = renderHook(() => useAnimationStyles());

        // Style should be added
        expect(document.getElementById(STYLE_ID)).toBeTruthy();

        // Unmount the hook
        unmount();

        // Style should still be present (no cleanup on unmount)
        expect(document.getElementById(STYLE_ID)).toBeTruthy();
    });

    it('should be idempotent across re-renders', () => {
        const { rerender } = renderHook(() => useAnimationStyles());

        const initialStyleEl = document.getElementById(STYLE_ID);
        expect(initialStyleEl).toBeTruthy();
        const initialContent = initialStyleEl?.textContent;

        // Re-render multiple times
        rerender();
        rerender();
        rerender();

        // Should still be the same element with same content
        const finalStyleEl = document.getElementById(STYLE_ID);
        expect(finalStyleEl).toBe(initialStyleEl); // Same element reference
        expect(finalStyleEl?.textContent).toBe(initialContent); // Same content

        // Should still only be one style element
        expect(document.querySelectorAll(`#${STYLE_ID}`).length).toBe(1);
    });
});
