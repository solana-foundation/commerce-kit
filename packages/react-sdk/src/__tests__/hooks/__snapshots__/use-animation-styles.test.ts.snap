// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useAnimationStyles > should inject complete animation styles with proper formatting 1`] = `
"@keyframes sc-tip-modal-slide-up {
  0% { transform: translateY(16px); opacity: 0; }
  100% { transform: translateY(0); opacity: 1; }
}
.sc-tip-modal-anim {
  transition: height 500ms cubic-bezier(0.4, 0, 0.2, 1) !important;
}
@media (prefers-reduced-motion: reduce) {
  .sc-tip-modal-anim { 
    animation: none !important;
    transition: none !important;
  }
}"
`;
